templates/apps/sumo-aws-apps/sumologic-app-utils/src/awsresource.py

```diff
@@ -31,7 +32,42 @@ def delete(self, *args, **kwargs):
     @abstractmethod
     def extract_params(self, event):
         pass
-
+class AWSValidation(AWSResource):
+    def __init__(self, props,  *args, **kwargs):
+        self.CLOUDFORMATION_PARAMETERS = ["PARAMETER_NAME", "PARAMETER_VALUE", "PARAMETER_ALLOWED_PATTERN"]
+    def create(self, params, *args, **kwargs):
+        try:
+            pat = re.compile(params["PARAMETER_ALLOWED_PATTERN"])
```

Comment:
instead of copying same code in create and update. Can we just create a validate function and call it from both the places with attributes

Reply:
fixed